insights/tests/client/connection/test_checkin.py

```diff
@@ -191,3 +198,22 @@ def test_response_failure(get_proxies, post, get_canonical_facts, rm_conf, statu
 
     with raises(Exception):
         connection.checkin()
+
+
+@patch('insights.client.connection.InsightsUploadConf.get_rm_conf', return_value={})
+@patch("insights.client.connection.InsightsConnection._init_session", Mock())
+@patch("insights.client.connection.InsightsConnection._fetch_system_by_machine_id", return_value=False)
+@patch("insights.client.connection.get_canonical_facts")
+@patch("insights.client.connection.InsightsConnection.post")
+@patch("insights.client.connection.InsightsConnection.get_proxies")
+def test_unregistered_host(get_proxies, post, get_canonical_facts, fetch_system_by_machine_id, rm_conf):
+    """
+    If the host is not registered, the checkin fails.
```

Comment:
I can imagine a bit more descriptive description like 

```suggestion
    If the host is not found in the inventory, the checkin doesn't happen.
```

This is a test, not a message for the user, so we can be more technically precise about what’s going on.

No need to block the PR because of this, however.